### PR TITLE
Download icon isn't turning green when clicking to download

### PR DESCRIPTION
**Pull request by mollfpr**
### Details

- Creating a HOC to hold the logic of button state complete
- Updating the `ContextMenuItem` to use the HOC and remove unused code
- Updating the `HeaderWithCloseButton` component to class-based
- Replacing the `TouchableOpacity` component with `Pressable` component
- Use the HOC on `HeaderWithCloseButton` and implement it to the download button

### Fixed Issues

$ https://github.com/Expensify/App/issues/9124

### Tests

1. Go to https://localhost:8080
2. Log in with any account and select any chat
3. Click on any attachment
4. On Preview page click download icon
5. Verify that the download icon turns green after click, and turns back to grey after a few seconds.

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose and it is
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps

1. Go to https://staging.new.expensify.com/
2. Log in with any account and select any chat
3. Click on any attachment
4. On Preview page click download icon
5. Verify that the download icon turns green after click, and turns back to grey after a few seconds.

- [x] Verify that no errors appear in the JS console

#### Web

https://user-images.githubusercontent.com/25520267/173381027-61c33f23-a15f-4798-b0c2-1fcb91b07fc7.mov

#### Mobile Web

https://user-images.githubusercontent.com/25520267/173381073-e1bfbfbe-be34-4cf7-a3f6-47688fe86642.mov

https://user-images.githubusercontent.com/25520267/173381098-b8780d6e-19cb-4aab-a739-3f9d1232d661.mov

#### Desktop

https://user-images.githubusercontent.com/25520267/173381163-5b5bbe1a-1b15-4bcf-8a56-9b4d67c07a46.mov

#### iOS

https://user-images.githubusercontent.com/25520267/173381225-729f77b4-76c3-4987-b8cf-4f45298c8c63.mov

#### Android

https://user-images.githubusercontent.com/25520267/173381255-a9f3199f-fb09-46a1-9d44-1cc060cd229c.mov

### CODE REVIEW

**Review by rushatgabhane — CHANGES_REQUESTED** [review]

File: src/components/withDelayToggleButtonState.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+
+const withDelayToggleButtonStatePropTypes = {
+    /** A value whether the button state is complete */
+    isButtonStateComplete: PropTypes.bool.isRequired,
```
Comment (rushatgabhane) [inline src/components/withDelayToggleButtonState.js:7]: NAB:  How about this?

`isButtonStateComplete` -> `isDelayButtonStateComplete`

File: src/components/withDelayToggleButtonState.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    /** A function to call to change the complete state */
+    toggleButtonStateComplete: PropTypes.func.isRequired,
```
Comment (rushatgabhane) [inline src/components/withDelayToggleButtonState.js:10]: NAB: how about this rename?

`toggleButtonStateComplete` -> `toggleDelayButtonState`

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
@mollfpr left some minor refactor comments

File: src/components/withDelayToggleButtonState.js (hunk 1 above)
Comment (mollfpr) [reply src/components/withDelayToggleButtonState.js:7]: Yup, that’s more logical!

**Comment by mollfpr** [comment]
Updated video for web and desktop.

**Comment by rushatgabhane** [comment]
@mollfpr Thanks for the rapid changes! Will review this in the evening (~6 hrs)

**Comment by rushatgabhane** [comment]
I'm confused by `withDelayToggleButtonState`. We can simplify it a bit. 
More soon

File: src/components/ContextMenuItem.js [hunk 3]
```diff
@@ -45,44 +48,26 @@ const defaultProps = {
...
         // We only set the success state when we have icon or text to represent the success state
         // We may want to replace this check by checking the Result from OnPress Callback in future.
         if (this.props.successIcon || this.props.successText) {
-            this.setState({
-                success: true,
-            });
-            if (this.props.autoReset) {
-                this.successResetTimer = setTimeout(() => this.setState({success: false}), 1800);
-            }
+            this.props.toggleDelayButtonState(this.props.autoReset);
```
Comment (rushatgabhane) [inline src/components/ContextMenuItem.js:64]: In context menu, we are never using the delay. So we should just use a state instead of the HOC.

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
@mollfpr I noticed that we aren't making use of the delay in context menu. So the HOC isn't needed in ContextMenu.

I've created a patch to rename a few things and removed the use of HOC by `ContextMenuItem`. Please apply this patch and feel free to make any additional changes. Curious for your thoughts




[patch here](https://github.com/Expensify/App/files/8861795/patch.txt) 
`git apply path/to/file/patch.txt`

File: src/components/ContextMenuItem.js (hunk 3 above)
Comment (mollfpr) [reply src/components/ContextMenuItem.js:64]: Thanks! I’ll apply the patch and try to take a look at the rest.

Comment (rushatgabhane) [reply src/components/ContextMenuItem.js:64]: Cool, yeah I missed `CommunicationsLink`.

**Comment by rushatgabhane** [comment]
@mollfpr the download button in attachment view isn't resetting it's state.

https://user-images.githubusercontent.com/29673073/173358619-4a351f50-6ae8-4521-83d4-299498660af9.mov

**Comment by rushatgabhane** [comment]
> We can pass the argument to the function to auto-reset the status

@mollfpr yes let's reset it's status in the AttachmentView.

> Are we need more changes

No, that's all :)

**Comment by mollfpr** [comment]
Updated video!

**Review by rushatgabhane — DISMISSED** [review]
@chiragsalian LGTM! 

<details>
<summary>PR Reviewer Checklist</summary>

- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

**Comment by rushatgabhane** [comment]
@mollfpr can you please modify the 5th point in tests to  ->  Verify that the download icon turns green after click, `and turns back to grey after a few seconds.`

**Comment by mollfpr** [comment]
Updated test steps!

File: src/components/ContextMenuItem.js [hunk 4]
```diff
@@ -45,44 +48,26 @@ const defaultProps = {
...
-        }
-
-        clearTimeout(this.successResetTimer);
+        this.triggerPressAndUpdateSuccess = this.triggerPressAndUpdateSuccess.bind(this);
     }
 
-    /**
-     * Called on button press and mark the run
-     */
     triggerPressAndUpdateSuccess() {
```
Comment (chiragsalian) [inline src/components/ContextMenuItem.js:55]: missing JS doc

File: src/components/HeaderWithCloseButton.js [hunk 5]
```diff
@@ -100,94 +106,119 @@ const defaultProps = {
...
-                        iconStyles={[styles.mr0]}
-                        anchorPosition={props.threeDotsAnchorPosition}
+class HeaderWithCloseButton extends Component {
+    constructor(props) {
+        super(props);
+
+        this.triggerButtonCompleteAndDownload = this.triggerButtonCompleteAndDownload.bind(this);
+    }
+
+    triggerButtonCompleteAndDownload() {
```
Comment (chiragsalian) [inline src/components/HeaderWithCloseButton.js:116]: missing js docs

File: src/components/withDelayToggleButtonState.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+                return;
+            }
+
+            clearTimeout(this.resetButtonStateCompleteTimer);
+        }
+
+        /**
+         * @param {Boolean} [resetAfterDelay] Impose delay before toggling state
+         */
+        toggleDelayButtonState(resetAfterDelay) {
```
Comment (chiragsalian) [inline src/components/withDelayToggleButtonState.js:35]: ~provide default value for optional param.~

Comment (chiragsalian) [reply src/components/withDelayToggleButtonState.js:35]: or rather why is the param doc optional. Looking at the code i don't see any flow where its optional so i think you can just remove `[resetAfterDelay]` from the JS doc.

**Review by chiragsalian — CHANGES_REQUESTED** [review]
Minor comments, rest LGTM and works well.

**Review by chiragsalian — CHANGES_REQUESTED** [review]

File: src/components/HeaderWithCloseButton.js [hunk 7]
```diff
@@ -100,94 +106,122 @@ const defaultProps = {
...
-                        anchorPosition={props.threeDotsAnchorPosition}
+class HeaderWithCloseButton extends Component {
+    constructor(props) {
+        super(props);
+
+        this.triggerButtonCompleteAndDownload = this.triggerButtonCompleteAndDownload.bind(this);
+    }
+
+    /**
+     * Called on download button press
```
[the remark this answers is not in the document]
Comment (mollfpr) [reply src/components/HeaderWithCloseButton.js:117]: Thank you for your suggestion. I’ll added it.

**Review by chiragsalian — APPROVED** [review]
LGTM.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [ ] Android / native
    - [x] iOS / Safari
    - [ ] Android / Chrome
    - [x] MacOS / Chrome
    - [ ] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn’t already exist
    - [x] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

**Comment by chiragsalian** [comment]
Merging since rushat approved it before and the changes are minimal since that review.

**Pull request merged**